### ISSUE TITLE
Dropdownlist - how to reset value and selection ?

### ISSUE THREAD

**Issue by zhao125787935** [telerik/kendo-angular#606]
Hi kendo team,

I have a button and a dropdown, when I click the button, I will have to reset the dropdown value to null and deselect the selection,   I try with following codes:  
                        let newValue: any = { value: null, name: "" };
			this.dropdown.value = newValue;
			this.dropdown.selectionChange.emit(newValue);
But it does not work, the previous selection is still there. And clicking the previous selection will not respond.
Seems that it does not trigger the selectionChange event.  Would you please kindly help with this issue?

Thanks

**Comment by valchev** [comment]
Hi @zhao125787935,

To reset the value, simply set it to `undefined`. The selection will automatically reset as it is bound to the value.

`this.dropdown.value = undefined;`

**Comment by zhao125787935** [comment]
hi @valchev 

I try with your approach, after set the value to be undefined,  it  will work when you click somewhere else except the dropdown,  seems that it triggers "blur " event.   then the value and selection will change.  Otherwise, the value will remain the same. 
Any ideas?

**Comment by valchev** [comment]
@zhao125787935 I prepared a [small example](http://plnkr.co/edit/Ck0pHP2kl9Pl0Km072I5?p=preview). Everything seems to work fine on my side, am I missing something?

**Comment by valchev** [comment]
I am closing the issue due to inactivity, feel free to reopen with more details.

In addition, we will shortly provide a `reset` method for all DropDown components that may be used to reset the component as setting the value to `undefined` or `null` does not work in all scenarios. For further details refer to telerik/kendo-angular#656

**Comment by josemolina001** [comment]
After making my initial selection with a filterable list of objects, upon re-opening the combobox I reset the data to the original list of data (non-filtered list). It keeps the selected index of the filtered list selection so upon re-opening it selects the incorrect data item. I have tried calling reset which does reset the selection. However, I'm having issues re-selecting the correct item. I'm able to data.find the correct data item but when I call this.element.selectionChange.emit(myObjectToSelect) it doesn't do anything. I tried setting this.element.value = myObjectToSelect, this.element.dataItem = myObjectToSelect, this.element.selected = myObjectToSelect but nothing works.

I'm doing this inside of a directive.
